lib/modules/manager/flux/extract.spec.ts

```diff
@@ -217,6 +230,137 @@ describe('modules/manager/flux/extract', () => {
       expect(result?.deps[0].skipReason).toBe('unknown-registry');
     });
 
+    it('ignores GitRepository without a tag nor a commit', () => {
+      const result = extractPackageFile(
+        codeBlock`
+          apiVersion: source.toolkit.fluxcd.io/v1beta1
+          kind: GitRepository
+          metadata:
+            name: renovate-repo
+            namespace: renovate-system
+          spec:
+            url: https://github.com/renovatebot/renovate
+        `,
+        'test.yaml'
+      );
+      expect(result?.deps[0].skipReason).toBe('unversioned-reference');
```

Comment:
please prefer to match full result object with toEqual, so things can't silently break

Reply:
Hi @viceice nice to see you back : )
I tried to change every `toBe` to `toEqual` and I got this error.
```
 > podman run -it --env-file .env --rm -v "$PWD":/usr/src/app -w /usr/src/app renovatebot_local yarn test
yarn run v1.22.19
$ run-s generate:*
$ node tools/generate-imports.mjs
generating imports
> data/azure-pipelines-tasks.json
> data/debian-distro-info.json
> data/kubernetes-api.json5
> data/node-js-schedule.json
> data/ubuntu-distro-info.json
> node_modules/emojibase-data/en/shortcodes/github.json
generating hashes
$ run-s lint test-schema type-check strict-check jest
$ run-s ls-lint eslint prettier markdown-lint git-check doc-fence-check
$ ls-lint
$ eslint . --cache --report-unused-disable-directives

/usr/src/app/lib/modules/manager/flux/extract.spec.ts
   88:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  111:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  158:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  178:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  199:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  214:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  230:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  246:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  264:45  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  265:45  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  284:44  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  303:44  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  322:44  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  341:44  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
  360:44  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be

✖ 15 problems (15 errors, 0 warnings)
  15 errors and 0 warnings potentially fixable with the `--fix` option.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "eslint" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "lint" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
So I applied the `lint-fix` and only some of the `toBe` have remained, and the line you mentioned is not one of them. 
```
  246:42  error  Use `toBe` when expecting primitive literals  jest/prefer-to-be
```
I git pulled and did `yarn install` so im kind of up to date with the main branch, did i miss a step that made the linter complain about this line?

Reply:
you should rewrite text like:

`expect(result).toEqual({ ... });`

so we've a single expectation.

Reply:
I updated my working branch with

```js
      expect(result?.deps[0]).toEqual({
        depName: 'renovate-repo',
        skipReason: 'unversioned-reference',
      });
```

but it seems like we need to target `result` in the expect directly